### PR TITLE
Don't skip file paths contains special words like `constructor`

### PR DESCRIPTION
**Pull request by ashlkv**
## Description

After updating to v2 prettier cli started throwing a "No matching files" error for paths like "js/constructor/*.js" in my team's project. It turned out that the `containsIgnoredPathSegment` check was returning a positive for the paths containing "constructor" segment, and was ignored.

This pull request fixes the issue by additionally checking the path segments against the silently ignored path dictionary with a `hasOwnProperty`.

## Checklist

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/cli/expand-patterns.js [hunk 1]
```diff
@@ -163,7 +163,11 @@ function containsIgnoredPathSegment(absolutePath, cwd, ignoredDirectories) {
   return path
     .relative(cwd, absolutePath)
     .split(path.sep)
-    .some((dir) => ignoredDirectories[dir]);
+    .some(
```
Comment (fisker) [inline src/cli/expand-patterns.js:166]: Can you pass a `Set` to this function instead of `object`?

Comment (fisker) [reply src/cli/expand-patterns.js:166]: I mean turn `silentlyIgnoredDirs` to `Set`

**Comment by fisker** [comment]
This bug is hilarious! 👍

**Comment by fisker** [comment]
@thorn0 I checked your recently committed code https://github.com/prettier/prettier/blob/a8f35961c066de440ceb31c9b7414c88cf91e7af/src/document/doc-debug.js#L184, glad you use `Object.create(null)`, haha

**Comment by thorn0** [comment]
@fisker The CLI part of the project isn't my cup of tea. :)

**Comment by fisker** [comment]
You are not alone, but this bug is on you, [it was `array`](https://github.com/prettier/prettier/pull/7660/files#diff-a1489d380e2e3986584e73d3f58312a0cb45e28853116312abc1d801474bd6b3L439), 😄

File: src/cli/expand-patterns.js (hunk 1 above)
Comment (ashlkv) [reply src/cli/expand-patterns.js]: Done in https://github.com/prettier/prettier/pull/10256/commits/8fbf36f72c2d1eb98f57039de54ffe1f83acfad3

**Review by fisker — APPROVED** [review]

**Comment by thorn0** [comment]
Changelog entry?

**Pull request merged**